src/actions/ws_message.rs

```diff
@@ -0,0 +1,109 @@
+use std::collections::HashMap;
+
+use yaml_rust::Yaml;
+use colored::*;
+use serde_json;
+use time;
+
+use ws::{connect, CloseCode};
+
+use interpolator;
+use config;
+
+use actions::{Runnable, Report};
+
+#[derive(Clone)]
+pub struct WSMessage {
+  name: String,
+  url: String,
+  time: f64,
+  data: Option<String>,
+  pub with_item: Option<Yaml>,
+  pub assign: Option<String>,
+}
+
+impl WSMessage {
+  pub fn is_that_you(item: &Yaml) -> bool{
+    item["websocket_message"].as_hash().is_some()
+  }
+
+  pub fn new(item: &Yaml, with_item: Option<Yaml>) -> WSMessage {
+    let reference: Option<&str> = item["assign"].as_str();
+    let data: Option<&str> = item["websocket_message"]["data"].as_str();
+
+    WSMessage {
+      name: item["name"].as_str().unwrap().to_string(),
+      url: item["websocket_message"]["url"].as_str().unwrap().to_string(),
+      time: 0.0,
+      data: data.map(str::to_string),
+      with_item: with_item,
+      assign: reference.map(str::to_string),
+    }
+  }
+
+  fn send_request(&self, context: &mut HashMap<String, Yaml>, responses: &mut HashMap<String, serde_json::Value>, _config: &config::Config) -> (f64) {
+
+    let begin = time::precise_time_s();
+
+    let interpolated_name;
+    let interpolated_url;
+
+    // Resolve the url
+    {
+      let interpolator = interpolator::Interpolator::new(context, responses);
+      interpolated_name = interpolator.resolve(&self.name);
+      interpolated_url = interpolator.resolve(&self.url);
+    }
+
+    let data_clone = match &self.data {
```

Comment:
It could be cool to interpolate data too

Reply:
Hmm, I'll check how that works. Can you give a brief tl;dr about what the purpose of the data interpolation is?